### PR TITLE
feat: Use extras to install specific dependencies

### PR DESCRIPTION
**Pull request by BoboTiG**
The only missing piece is MQTT, it is way to intriqued everywhere.
So for now, the `paho-mqtt` dependency will be installed by default.

At some point, it will be cool to be able to get rid of it and install it via:

```shell
$ python -m pip install 'mppsolar[mqtt]'
```

Also, I added very primary tests to see if all protocols and outputs can be
imported even when a depedency is missing.

A proper message will be printed
in that case (except for `esp32io.py` and the `machine` import, I guss this is very specific?).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,7 +21,12 @@ Currently has support for:
 
 ## Install ##
 - `pip install mppsolar` (minimal install), or
-- `pip install mppsolar[ble]` (install including BLE support aka jkbms), or
+- `pip install mppsolar[api]` (install server stuff), or
```
Comment (BoboTiG) [inline README.md:24]: I am not proud about "server stuff", so if you come up with something better, that would be appreciated :)

File: tests/mppsolar_tests/test_outputs.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import importlib
+import unittest
+from pathlib import Path
+
+HERE = Path( __file__).parent
+OUTPUTS_DIR = HERE.parent.parent / "mppsolar" / "outputs"
+OUTPUTS = sorted(
+    file.stem for file in OUTPUTS_DIR.glob("*.py")
+    if file.stem != "__init__"
+)
```
Comment (BoboTiG) [inline tests/mppsolar_tests/test_outputs.py:10]: We can't use `list_output()` here, for two reasons:
- it doesn't work from tests, it seems the fact that it's not packaged is messing the whole logic
- more importantly, we need to really try all outputs, without skipping ones that are not available

Same remark for the protocol test file below.

**Comment by gianfrdp** [comment]
This PR breaks my previous PR #428

for file mppsolar/protocols/abstractprotocol.py

firefox_2023-12-15_15-17-37

**Comment by BoboTiG** [comment]
Then a `# noqa: F401` would have been wise. I'll open a PR to ix that.

**Pull request merged**